docs/guide/helper-array.md

```diff
@@ -312,7 +312,8 @@ or an anonymous function like the following one:
 
 ```php
 ArrayHelper::multisort($data, function($item) {
-    return isset($item['age']) ? ['age', 'name'] : 'name';
```

Comment:
Wasn't it meant to sort by age and then by name?

Reply:
No, it will generate $args that are passed to `array_multisort` like this:
```php
array (
  0 => 
  array (
    0 => 
    array (
      0 => 'age',
      1 => 'name',
    ),
    1 => 
    array (
      0 => 'age',
      1 => 'name',
    ),
    2 => 
    array (
      0 => 'age',
      1 => 'name',
    ),
  ),
  1 => 4,
  2 => 0,
  3 => 
  array (
    0 => 1,
    1 => 2,
    2 => 3,
  ),
  4 => 4,
  5 => 1,
  6 => 
  array (
    0 => 
    array (
      'age' => 30,
      'name' => 'Alexander',
    ),
    1 => 
    array (
      'age' => 19,
      'name' => 'Barney',
    ),
    2 => 
    array (
      'age' => 30,
      'name' => 'Brian',
    ),
  ),
)
```

See the first element `$args[0]`.

Reply:
@samdark am I missing something?

Reply:
Yes. From phpdoc:

> Sorts an array of objects or arrays (with the same structure) by one or **several** keys.

> * @param string|\Closure|array $key the key(s) to be sorted by. This refers to a key name of the sub-array elements, a property name of the objects, or an anonymous function returning the values for comparison purpose. The anonymous function signature should be: `function($item)`.
**To sort by multiple keys, provide an array of keys here.**

Reply:
See example 3: https://www.php.net/manual/en/function.array-multisort.php

Reply:
@samdark yes, I saw this example. 
Have you seen my comments here? https://github.com/yiisoft/yii2/issues/17372#issuecomment-502648952
The problem is that anonymous function that is passed to the `ArrayHelper::multisort()` is not used to get keys for sorting but it is passed to the `\yii\helpers\BaseArrayHelper::getColumn` to get modified values of the sorting array. It serves a completely different purpose in the current implementation.

> To sort by multiple keys, provide an array of keys here.
If you pass an array of keys to the `ArrayHelper::multisort()` it works correctly because it builds correct `$args` for the `array_multisort`:

```php
ArrayHelper::multisort($data, ['age', 'name'], [SORT_ASC, SORT_DESC]);
```

```php
// $args
array (
  0 => 
  array (
    0 => 30,
    1 => 30,
    2 => 19,
  ),
  1 => 4,
  2 => 0,
  3 => 
  array (
    0 => 'Alexander',
    1 => 'Brian',
    2 => 'Barney',
  ),
  4 => 3,
  5 => 0,
  ...
)
```

But if you pass an anonymous function it will generate `$args` that is completely different and doesn't make any sens:

```php
        ArrayHelper::multisort($data, function ($item) {
            return ['age', 'name'];
        }, SORT_DESC);
```

```php
// $args
array (
  0 => 
  array (
    0 => 
    array (
      0 => 'age',
      1 => 'name',
    ),
    1 => 
    array (
      0 => 'age',
      1 => 'name',
    ),
    2 => 
    array (
      0 => 'age',
      1 => 'name',
    ),
  ),
  1 => 4,
  2 => 0,
)
```

`ArrayHelper::multisort()` is a bit messy, so that is why it's hard to understand and explain what is going on.

Reply:
What a mess...